### ISSUE TITLE
generate-xorg-conf.sh generates broken xorg.conf

### ISSUE THREAD

**Issue by jgroenveld** [project-trident/trident-core#44]
```
There's a couple bugs with trident-core-201811161350 in the generated xorg.conf :
1. It selects the VESA vice modesetting due to the ancient host firmware only supporting 
BIOS mode.

*** /root/detect-best-driver.sh.trident Tue Nov 27 10:41:54 2018
--- /usr/local/share/trident/scripts/detect-best-driver.sh      Tue Nov 27 10:40:38 2018
***************
*** 29,35 ****
    if [ "${efi}" = "UEFI" ] ; then
      driver="modesetting intel"
    else
!     driver="intel"
    fi
  
  elif [ "${vendor_detected}" = "nvidia" ] ; then
--- 29,35 ----
    if [ "${efi}" = "UEFI" ] ; then
      driver="modesetting intel"
    else
!     driver="modesetting intel"
    fi
  
  elif [ "${vendor_detected}" = "nvidia" ] ; then


2. The Optimus detection selects nVidia vice the Intel VGA device due to the above driver detection
routine assuming modesetting only working with EFI.

John
groenveld@acm.org
```
[xorg.conf.corrected.txt](https://github.com/project-trident/trident-core/files/2620163/xorg.conf.corrected.txt)
[xorg.conf.generated.txt](https://github.com/project-trident/trident-core/files/2620162/xorg.conf.generated.txt)
[pciconf.txt](https://github.com/project-trident/trident-core/files/2620143/pciconf.txt)
[dmidecode.txt](https://github.com/project-trident/trident-core/files/2620144/dmidecode.txt)
[devinfo.txt](https://github.com/project-trident/trident-core/files/2620145/devinfo.txt)

**Comment by beanpole135** [comment]
The requested change is basically asking for the modesetting driver to be automatically used for non-UEFI systems.
I am 90% sure that either the modesetting driver or the drm-current-kmod package specifically states that it should only be used on UEFI systems in order to leverage the KMS hooks.

**Comment by jgroenveld** [comment]
```
This particular Optimus laptop is ancient.
dmidecode shows 2012.
# dmidecode 3.2
Scanning /dev/mem for entry point.
SMBIOS 2.7 present.
33 structures occupying 2109 bytes.
Table at 0xC9EC2018.

Handle 0x0000, DMI type 0, 24 bytes
BIOS Information
        Vendor: American Megatrends Inc.
        Version: 4.6.5
        Release Date: 05/16/2012
        Address: 0xF0000
        Runtime Size: 64 kB
        ROM Size: 4096 kB
        Characteristics:
                PCI is supported
                BIOS is upgradeable
                BIOS shadowing is allowed
                Boot from CD is supported
                Selectable boot is supported
                BIOS ROM is socketed
                EDD is supported
                5.25"/1.2 MB floppy services are supported (int 13h)
                3.5"/720 kB floppy services are supported (int 13h)
                3.5"/2.88 MB floppy services are supported (int 13h)
                Print screen service is supported (int 5h)
                8042 keyboard services are supported (int 9h)
                Serial services are supported (int 14h)
                Printer services are supported (int 17h)
                ACPI is supported
                USB legacy is supported
                BIOS boot specification is supported
                Targeted content distribution is supported
                UEFI is supported
        BIOS Revision: 4.6

UEFI may be supported, but I'll be damned if I can find a BIOS menu option to enable it.
Garbage firmware makes garbage OS!
So long as trident_init respects my /etc/X11/xorg.conf, I'm inclined to close.
Suggest waiting to see if others stumble with similar vintage hardware.
John
groenveld@acm.org
```

**Comment by beanpole135** [comment]
Fixed with aef483f8296c469594c58cd2c52497d9a8dc3e96
I added the modetting driver to BIOS-boot systems, but have a failover system in place to use the vesa or scfb driver in case the modesetting driver fails when starting up X11.
